fe/fe-core/src/main/java/com/starrocks/sql/common/TypeManager.java

```diff
@@ -104,18 +104,25 @@ private static Type getCommonMapType(MapType t1, MapType t2) {
     }
 
     private static Type getCommonStructType(StructType t1, StructType t2) {
```

Comment:
the method will modify cast logical, I think should need describe `what are you doing` and add more test case @wangsimo0 @yingtingdong struct common type change.
the common type of `struct<a int, b bigint>` and `struct<a bigint, b int, c int>` is `struct<a bigint, b bigint, c int>` is

Reply:
@Seaven Thanks, I updated `What I'm doing`

Reply:
I get your intention, but I don’t recommend changing the code `getCommonStructType`, the method will effect `cast(row(1,2) as struct<a int, b int, c int>)`, and BE doesn't support now.

If you only want to fix `IF(xx, NULL, struct)` bug, I think you can handle it in `PolymorphicFunctionAnalyzer.resolveByDeducingReturnType`, to handle NULL by a speical process

Reply:
@Seaven Good suggestion, I will optimize it again.